### PR TITLE
Pull Request Metrics POC

### PR DESCRIPTION
**Pull request by patricktnast**
### Description

- *Category*: 
POC
- *JIRA issue*: https://jira.ihme.washington.edu/browse/MIC-5114

### Changes and notes
 
This is the first draft of a workflow that tests pull review metrics for various framework repos. Rajan suggested I do this in this repo. Unfortunately, it seems like i have to have *some* workflow merged to main to be able to run this, so I can't test it until then. I anticipate some further tweaks, but this should work more or less.

It grabs prs closed in the last two weeks (~last sprint) as well as currently open prs with some statistics on the time they spent in flight. 
### Testing

none

### CODE REVIEW

File: .github/workflows/pr_metrics.yml [hunk 1]
```diff
@@ -0,0 +1,59 @@
+name: Post-Sprint PR metrics
+on:
+  workflow_dispatch:
```
Comment (albrja) [inline .github/workflows/pr_metrics.yml:3]: You could change this to be a cron job every two weeks but if we ever moved the meeting wed have to update this.

**Review by albrja — APPROVED** [review]

File: .github/workflows/pr_metrics.yml (hunk 1 above)
Comment (stevebachmeier) [inline .github/workflows/pr_metrics.yml:3]: isn't this missing something? I don't see a value of `workflow_dispatch`

File: .github/workflows/pr_metrics.yml [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+          
+          # Set an environment variable with the date range
+          echo "$fourteen_days_ago..$yesterday"
+          echo "last_two_weeks=$fourteen_days_ago..$yesterday" >> "$GITHUB_ENV"
+
+      - name: Get issue metrics for PRs closed in last sprint
+        uses: github/issue-metrics@v3
+        env:
+          GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          SEARCH_QUERY: $${{ env.REPO_LIST }} is:pr closed:${{ env.last_two_weeks }} -reason:"not planned"'
```
Comment (stevebachmeier) [inline .github/workflows/pr_metrics.yml:37]: what does `-reason:"not planned"` do?

File: .github/workflows/pr_metrics.yml [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+
+      - name: Create issue for PRs closed in last sprint
+        uses: peter-evans/create-issue-from-file@v4
+        with:
+          title: Post-Sprint Closed Pull Request Metrics Report (dev)
+          token: ${{ secrets.GITHUB_TOKEN }}
+          content-filepath: ./issue_metrics.md
+          assignees: patricktnast
```
Comment (stevebachmeier) [inline .github/workflows/pr_metrics.yml:45]: Probably better to use the svc-simsci user

File: .github/workflows/pr_metrics.yml [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+      - name: Get issue metrics for currently open PRs
+        uses: github/issue-metrics@v3
+        env:
+          GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          SEARCH_QUERY: '$${{ env.REPO_LIST }} is:pr is:open'
+
+      - name: Create issue for currently open PRs
+        uses: peter-evans/create-issue-from-file@v4
+        with:
+          title: Post-Sprint Closed Pull Request Metrics Report (dev)
```
Comment (stevebachmeier) [inline .github/workflows/pr_metrics.yml:56]: should this be "Post-Sprint **_Open_** Pull Request ..."

File: .github/workflows/pr_metrics.yml [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+          content-filepath: ./issue_metrics.md
+          assignees: patricktnast
+          
+      - name: Get issue metrics for currently open PRs
+        uses: github/issue-metrics@v3
+        env:
+          GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          SEARCH_QUERY: '$${{ env.REPO_LIST }} is:pr is:open'
+
+      - name: Create issue for currently open PRs
```
Comment (stevebachmeier) [inline .github/workflows/pr_metrics.yml:53]: This is identical to the previous create-issue step. Does it use `SEARCH_QUERY` in the background and that's how it's different?

**Review by stevebachmeier — APPROVED** [review]

**Comment by stevebachmeier** [comment]
Do you have an example of what we're expecting the output of this to actually look like? Guess we'll see when this gets merged

File: .github/workflows/pr_metrics.yml (hunk 1 above)
Comment (patricktnast) [reply .github/workflows/pr_metrics.yml:3]: Yeah, i was probably going to set this up once I know it works the way its supposed to. OTOH, just manually generating it at the retro might be the most failsafe thing to do for the reason you mentioned.

Comment (patricktnast) [reply .github/workflows/pr_metrics.yml:3]: I don't think it needs one? cf. https://github.com/marketplace/actions/issue-metrics

File: .github/workflows/pr_metrics.yml (hunk 4 above)
Comment (patricktnast) [reply .github/workflows/pr_metrics.yml:56]: good catch, ty!

File: .github/workflows/pr_metrics.yml (hunk 5 above)
Comment (patricktnast) [reply .github/workflows/pr_metrics.yml:53]: I think that is so, it uses the metrics from the most previous "github/issue-metrics@v3". I structured this according to :
https://github.com/github/issue-metrics/blob/main/docs/search-query.md

File: .github/workflows/pr_metrics.yml (hunk 3 above)
Comment (patricktnast) [reply .github/workflows/pr_metrics.yml:45]: Yeah, i was going to adjust how it reported to the team after I played with the workflow a bit more and didn't want to spam.

File: .github/workflows/pr_metrics.yml (hunk 2 above)
Comment (patricktnast) [reply .github/workflows/pr_metrics.yml:37]: i just copied it from the template TBH but looking into it more I think it's proably not appropriate for PRs

**Comment by patricktnast** [comment]
> Do you have an example of what we're expecting the output of this to actually look like? Guess we'll see when this gets merged

It will look something like the sample in https://github.com/marketplace/actions/issue-metrics
but I can't try until the workflow exists in GH.

**Pull request merged**